src/libraries/System.Private.CoreLib/src/System/Threading/Thread.cs

```diff
@@ -72,6 +88,13 @@ private void RunWorker()
 
                 try
                 {
+#if TARGET_OSX
+                    if (!string.IsNullOrEmpty(_thread.Name))
+                    {
+                        // Name the underlying native thread to match the managed thread name.
+                        _thread.ThreadNameChanged(_thread.Name);
```

Comment:
`ThreadNameChanged` sets the OS thread name, and also notifies profiler and debugger on CoreCLR about the thread name change. These notifications are superfluous here.